documentation/docs/protocol_specification/components/congestion_control.md

```diff
@@ -1,87 +1,173 @@
-    - since buffers are of a limited size, if the total number of bytes in all queues exceeds a certain threshold, new incoming messages are dropped;
-    - to guarantee the security of the network, if a certain queue exceeds a given threshold, new incoming packets from that specific node ID will be dropped.
-
-The dequeue mechanism includes the following components:
-
-*   _Queue selection_. A queue is selected according to round robin scheduling algorithm. In particular, we use a modified version of the deficit round robin (DRR) algorithm.
-*   _Message dequeuing_. The first message of the queue is dequeued, and list of active nodes is updated.
-*   _Scheduler management_. Scheduler counters and pointers are updated.
+* __Classification__: The mechanism identifies the queue where the message belongs according to the node ID of
+  the message issuer.
+* __Message enqueuing__: The message is actually enqueued, the queue is sorted by message timestamps in increasing order
+  and counters are updated (e.g., counters for the total number of bytes in the queue).
+* __Message skipping__. A message is skipped when it is confirmed while still waiting to be scheduled in the
+  outbox. A skipped message is removed from the outbox buffer, and it does not decrease the issuing node's deficit.
+  It is not gossipped or added to the tip pool as confirmation indicates that the message already has approvers and
+  is replicated on enough nodes in the network.
+* __Message drop__: In some circumstances, due to network congestion or ongoing attacks, some messages shall be dropped
+  to guarantee bounded delays and isolate the attacker's messages. Specifically, a node shall drop messages in the
+  following situations:
+    - Since the total outbox buffer has a limited size, if the total number of bytes in all queues exceeds the current
+      buffer size is smaller than the maximum size. The Mana scaled length of a queue is calculated by dividing the number
+      of bytes in the queue by the amount of access Mana of the corresponding node:
+    - `Mana-scaled queue size = queue size / node aMana`;
+
+The dequeuing mechanism includes the following components:
+
+* __Queue selection__: A queue is selected according to a round-robin scheduling algorithm. In particular, the 
+mechanism uses a modified version of the deficit round-robin (DRR) algorithm.
+* __Message dequeuing__. The first (oldest) message of the queue, that satisfies certain conditions is dequeued. A 
+  message must satisfy the following conditions:
+    * The message has a ready flag assigned. Ready flag is assigned to a message when all of its parents are eligible (parents have been scheduled or confirmed).
+    * The message timestamp is not in the future.
+* __Scheduler management__: The scheduler counters and pointers are updated.
+
+#### False positive drop
```

Comment:
Basically we want to say here that false positive drop is not a problem because dropped messages in the outbox are already booked and confirmation comes eventually due to messages received from the rest of the network which approve the dropped ones.

Reply:
Changed the description slightly towards the suggestion.